llvm/lib/IR/Verifier.cpp

```diff
@@ -5235,6 +5235,23 @@ void Verifier::visitIntrinsicCall(Intrinsic::ID ID, CallBase &Call) {
     }
     break;
   }
+  case Intrinsic::ucmp:
+  case Intrinsic::scmp: {
+
+    Type *SrcTy = Call.getOperand(0)->getType();
+    Type *DestTy = Call.getType();
+
+    Check(DestTy->getScalarSizeInBits() >= 2, "DestTy must be at least 2 bits wide", Call);
+
+    bool isDestTypeVector = DestTy->isVectorTy();
+    Check(SrcTy->isVectorTy() == isDestTypeVector,
+        "[u]scmp source and destination must both be a vector or neither", Call);
```

Comment:
```suggestion
        "[us]cmp source and destination must both be a vector or neither", Call);
```
? Though maybe I'd just drop the first word entirely...

Reply:
I'd like keeping that context if possible.